runtime/near-vm-runner/src/wasmer1_runner.rs

```diff
@@ -286,6 +286,7 @@ pub(crate) fn default_wasmer1_store() -> Store {
 pub(crate) fn run_wasmer1_module<'a>(
     module: &Module,
     store: &Store,
+    memory: &mut Wasmer1Memory,
```

Comment:
Wasmer memory is ref-counted object, so passing an exclusive reference to shared data doesn't make much sens. I'd make this accept memory by value. That should also simplify the calling code, as that no longer needs to be `&mut`

Reply:
I think this PR would be better to rebase on #4127 now. Also, we should have a benchmark on this PR to see reuse actually saving non-trivial time